syntax = "proto3";
import "google/protobuf/empty.proto";

message RawMeteoData {
 float temperature = 1;
 float humidity = 2;
 string timestamp = 3;
 //int sensorID;
}
message RawPollutionData {
 float co2 = 1;
 string timestamp = 2;
 //int sensorID;
}

message ProcessedData {
  float well = 1;
  float poll = 2;
  string timestampWell = 3;
  string timestampPoll = 4; 
}

service LBService {
  rpc AddAirData (RawMeteoData) returns (google.protobuf.Empty) {}
  rpc AddPollData (RawPollutionData) returns (google.protobuf.Empty) {}
}

service MeteoService {
  //rpc GetInsults (google.protobuf.Empty) returns (Insults) {}
  rpc ProcessMeteoData (RawMeteoData) returns (google.protobuf.Empty) {}
  rpc ProcessPollData (RawPollutionData) returns (google.protobuf.Empty) {}
  //rpc InsultMe (google.protobuf.Empty) returns (Insult) {}
}

service TerminalService {
  rpc AddTerminalData(ProcessedData) returns (google.protobuf.Empty) {}
}